integrations/datadog/main.py

```diff
@@ -0,0 +1,78 @@
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+
+from client import DatadogClient
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+class ObjectKind(StrEnum):
+    HOST = "host"
+    MONITOR = "monitor"
+    SLO = "slo"
+    SERVICE = "serviceCatalog"
+
+
+def init_client() -> DatadogClient:
+    return DatadogClient(
+        ocean.integration_config["datadog_base_url"],
+        ocean.integration_config["datadog_api_key"],
+        ocean.integration_config["datadog_application_key"],
+    )
+
+
+@ocean.on_resync()
+async def on_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dd_client = init_client()
+
+    if kind == ObjectKind.HOST:
```

Comment:
why is this using the same function for all kinds if you still have to implement a method per kind?

As I see it, either you implement a generic code that knows to bring the data just by the kind allowing the users to add new datadog kinds without any code intervention on our side. see azure integration for example.
Or if not possible due to more complex logic or not generic api from the third party provider then implement a resync method per kind.

Reply:
I understand. The api is not generic so I had to use different methods. I would have preffered to use one reysnc method but I see your concept so I will change to a method per kind